PWGHF/Core/HfMlResponseDplusToPiKPi.h

```diff
@@ -0,0 +1,167 @@
+        CHECK_AND_FILL_VEC(decayLength);
+        CHECK_AND_FILL_VEC(decayLengthXYNormalised);
+        CHECK_AND_FILL_VEC(cpa);
+        CHECK_AND_FILL_VEC(cpaXY);
+        CHECK_AND_FILL_VEC(maxNormalisedDeltaIP);
+        // TPC PID variables
+        CHECK_AND_FILL_VEC_FULL(prong0, tpcNSigmaPi0, tpcNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong0, tpcNSigmaKa0, tpcNSigmaKa);
+        CHECK_AND_FILL_VEC_FULL(prong1, tpcNSigmaPi1, tpcNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong1, tpcNSigmaKa1, tpcNSigmaKa);
+        CHECK_AND_FILL_VEC_FULL(prong2, tpcNSigmaPi2, tpcNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong2, tpcNSigmaKa2, tpcNSigmaKa);
+        // TOF PID variables
+        CHECK_AND_FILL_VEC_FULL(prong0, tofNSigmaPi0, tofNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong0, tofNSigmaKa0, tofNSigmaKa);
+        CHECK_AND_FILL_VEC_FULL(prong1, tofNSigmaPi1, tofNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong1, tofNSigmaKa1, tofNSigmaKa);
+        CHECK_AND_FILL_VEC_FULL(prong2, tofNSigmaPi2, tofNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong2, tofNSigmaKa2, tofNSigmaKa);
+        // Combined PID variables
+        CHECK_AND_FILL_VEC_FULL(prong0, tpcTofNSigmaPi0, tpcTofNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong1, tpcTofNSigmaPi1, tpcTofNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong2, tpcTofNSigmaPi2, tpcTofNSigmaPi);
+        CHECK_AND_FILL_VEC_FULL(prong0, tpcTofNSigmaKa0, tpcTofNSigmaKa);
+        CHECK_AND_FILL_VEC_FULL(prong1, tpcTofNSigmaKa1, tpcTofNSigmaKa);
+        CHECK_AND_FILL_VEC_FULL(prong2, tpcTofNSigmaKa2, tpcTofNSigmaKa);
+      }
+    }
+
+    return inputFeatures;
+  }
+
+ protected:
+  /// Method to fill the map of available input features
+  void setAvailableInputFeatures()
+  {
+    MlResponse<TypeOutputScore, EnumInputFeatures>::mAvailableInputFeatures = {
+      FILL_MAP(ptProng0),
+      FILL_MAP(ptProng1),
+      FILL_MAP(ptProng2),
+      FILL_MAP(impactParameterXY0),
+      FILL_MAP(impactParameterXY1),
+      FILL_MAP(impactParameterXY2),
+      FILL_MAP(decayLength),
+      FILL_MAP(decayLengthXYNormalised),
+      FILL_MAP(cpa),
+      FILL_MAP(cpaXY),
+      FILL_MAP(maxNormalisedDeltaIP),
+      // TPC PID variables
+      FILL_MAP(tpcNSigmaPi0),
+      FILL_MAP(tpcNSigmaKa0),
+      FILL_MAP(tpcNSigmaPi1),
+      FILL_MAP(tpcNSigmaKa1),
+      FILL_MAP(tpcNSigmaPi2),
+      FILL_MAP(tpcNSigmaKa2),
+      // TOF PID variables
+      FILL_MAP(tofNSigmaPi0),
+      FILL_MAP(tofNSigmaKa0),
+      FILL_MAP(tofNSigmaPi1),
+      FILL_MAP(tofNSigmaKa1),
+      FILL_MAP(tofNSigmaPi2),
+      FILL_MAP(tofNSigmaKa2),
+      // Combined PID variables
+      FILL_MAP(tpcTofNSigmaPi0),
+      FILL_MAP(tpcTofNSigmaPi1),
+      FILL_MAP(tpcTofNSigmaPi2),
+      FILL_MAP(tpcTofNSigmaKa0),
+      FILL_MAP(tpcTofNSigmaKa1),
+      FILL_MAP(tpcTofNSigmaKa2)};
+  }
+};
+
+} // namespace o2::analysis
+
+#undef FILL_MAP
+#undef CHECK_AND_FILL_VEC_FULL
+#undef CHECK_AND_FILL_VEC
```

Comment:
This does not look safe to me. What happens if we use different similar classes, e.g. for different particles, in the same file? Does it still work?

Reply:
Indeed, this I don't know.

Reply:
@fcatalan92 What is the issue with different particles? Why could the same macros not be used for all particles and undefined at the end of the file?

Reply:
@vkucera The point is that these macros are defined and undefined in different headers. So, if we imagine a task using the ML inference for two different particles we get:
```C++
#include "PWGHF/Core/HfMlResponse.h"  // not needed but for the sake of clarity
#include "PWGHF/Core/HfMlResponseDplusToPiKPi.h"
#include "PWGHF/Core/HfMlResponseOtherParticle.h"
```
This means that the macros are defined one time (due to the include guard) and undefined two times. I don't know if this would break compilation, but it looks potentially problematic.

Reply:
One has to undefine the macros after the last use, of course, so in this case it would be after `#include "PWGHF/Core/HfMlResponseOtherParticle.h"` in the `cxx` file.

Reply:
But since that would lead to a less clean code, I'm happy with the current version.